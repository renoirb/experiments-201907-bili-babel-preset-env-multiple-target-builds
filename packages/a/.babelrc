{
  "presets": [["@frontend-bindings/bundling-config/bili-babel"]],
}
