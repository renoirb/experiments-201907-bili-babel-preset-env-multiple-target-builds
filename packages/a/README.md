# `@frontend-bindings/a`

> TODO: description

## Usage

```
const a = require('@frontend-bindings/a');

// TODO: DEMONSTRATE API
```
