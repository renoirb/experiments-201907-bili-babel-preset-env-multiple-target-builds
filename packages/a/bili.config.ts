import { bili } from '@frontend-bindings/bundling-config'
import { Config, ConfigOutput, BabelPresetOptions } from 'bili'
import pkg from './package'

const output: ConfigOutput = {
  sourceMap: false,
  target: 'node',
}

const babel: BabelPresetOptions = {}

const config: Config = {
  babel,
  input: 'src/index.ts',
  output,
}

export default {
  ...config,
  ...bili.config(pkg),
}
