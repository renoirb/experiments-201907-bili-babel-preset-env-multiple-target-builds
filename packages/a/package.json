{
  "name": "@frontend-bindings/a",
  "version": "1.0.0",
  "description": "How much a could a fid in as",
  "homepage": "https://cgiraptor.atlassian.net/wiki/spaces/UN/pages/612204555/packages/a",
  "repository": {
    "type": "git",
    "url": "https://git.unify360.cgi.com:50043/unify360/frontend-bindings.git"
  },
  "license": "UNLICENSED",
  "files": [
    "dist"
  ],
  "main": "dist",
  "module": "dist/index.esm.js",
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "dependencies": {
    "@babel/runtime": "^7.2.0"
  },
  "publishConfig": {
    "registry": "https://nexus.unify360.cgi.com:50043/repository/npm-hosted/"
  },
  "copyright.owner": "CGI Inc."
}
