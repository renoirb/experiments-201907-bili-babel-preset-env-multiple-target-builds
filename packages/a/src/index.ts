export interface AlphaInterface {
  readonly name: string
  age: number
}

export class Alpha implements AlphaInterface {
  public age: number = 0
  constructor(public readonly name: string) {}
}
