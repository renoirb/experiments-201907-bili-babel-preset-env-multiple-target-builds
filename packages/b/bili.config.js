const { bili } = require('@frontend-bindings/bundling-config')
const pkg = require('./package')

const input = {
  index: 'src/index.js',
  cli: 'src/cli.js',
}

/** @type {import('bili').ConfigOutput} */
const output = {
  format: ['esm', 'cjs'],
}

/** @type {import('bili').Config} */
module.exports = {
  input,
  output,
  ...bili.config(pkg),
}
