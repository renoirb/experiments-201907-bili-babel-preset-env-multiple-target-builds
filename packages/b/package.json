{
  "name": "@frontend-bindings/b",
  "version": "1.0.0",
  "description": "Example of a ECMAScript module",
  "repository": {
    "type": "git",
    "url": "https://github.com/renoirb/experiments-201907-bili-babel-preset-env-multiple-target-builds"
  },
  "license": "UNLICENSED",
  "files": [
    "dist"
  ],
  "main": "dist",
  "module": "dist/index.esm.js",
  "scripts": {
    "build": "yarn run build:node",
    "build:node": "bili --verbose --stack-trace --target node",
    "prepublishOnly": "yarn run build",
    "test": "jest --verbose --maxWorkers=3 --detectOpenHandles",
    "test:watch": "jest --verbose --watch --maxWorkers=3 --detectOpenHandles"
  },
  "dependencies": {
    "@babel/runtime": "^7.4.0",
    "@frontend-bindings/a": "^1.0.0"
  },
  "devDependencies": {
    "@frontend-bindings/bundling-config": "^1.0.0",
    "lodash-es": "^4.17.10"
  }
}
