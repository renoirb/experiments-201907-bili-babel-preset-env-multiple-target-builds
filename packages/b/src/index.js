export * from './is-string'
export * from './sample-using-lodash'
export * from './sample-using-async'
import * as convert from './convert'
export * from './alpha'

export { convert }
