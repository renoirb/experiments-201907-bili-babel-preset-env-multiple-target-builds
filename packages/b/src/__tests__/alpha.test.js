import { createAlpha } from '@frontend-bindings/b'
import { Alpha } from '@frontend-bindings/a'

describe('alpha', () => {
  test('accessibility', () => {
    expect(createAlpha).toBeDefined()
  })

  test.skip('constructor', () => {
    // This is just an attempt at importing from more than one
    // module.
    expect(() => {
      new Alpha()
    }).toThrowError()
  })

  test('Snapshot', () => {
    const expectedName = 'Renoir Boulanger'
    const subject = createAlpha(expectedName)
    expect(subject).toMatchSnapshot()
    expect(subject).toHaveProperty('age', 0)
  })
})
