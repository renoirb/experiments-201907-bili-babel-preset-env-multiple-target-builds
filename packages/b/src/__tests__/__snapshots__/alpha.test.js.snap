// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alpha Snapshot 1`] = `
Alpha {
  "age": 0,
  "name": "Renoir Boulanger",
}
`;
